### PR TITLE
refactor(ME-1059): refactor tunnel connect

### PR DESCRIPTION
**Pull request by btoonk**
# Description

* refactor tunnel connect
* remove duplicate tunnel connect code
* remove sshkey/identity support for tunnels
* remove protected flags (not supported anymore)
* remove cloudAuthentication fields
* deprecated tunnel command
* deprecated connect command

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Tested locally, but needs more testing.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by pedroandrade — APPROVED** [review]
Hey @btoonk, there are a lot of changes, a good test will be really nice before merge

**Comment by adrianosela** [comment]
This is awesome, thanks @btoonk !

File: cmd/account.go [hunk 1]
```diff
@@ -128,17 +128,7 @@ var createCmd = &cobra.Command{
...
-				dat, err := os.ReadFile(sshkey)
-				if err != nil {
-					log.Fatalf("Unable to read the file %s, please check file permissions and try again (%v)", sshkey, err)
-				}
-				sshkey = string(dat)
-			}
-		}
-
-		err := http.Register(name, email, password, sshkey)
+		err := http.Register(name, email, password)
```
Comment (adrianosela) [inline cmd/account.go:131]: NOOP: We really need to get a better name for this `http` package...

File: internal/http/handler.go [hunk 2]
```diff
@@ -134,7 +134,7 @@ func (c *Client) Request(method string, url string, target interface{}, data int
 	}
 
 	if resp.StatusCode < http.StatusOK || resp.StatusCode > http.StatusNoContent {
-		responseData, err := ioutil.ReadAll(resp.Body)
+		responseData, err := io.ReadAll(resp.Body)
```
Comment (adrianosela) [inline internal/http/handler.go:137]: thank you!

**Review by adrianosela — APPROVED** [review]
Very nice. +1 on a good test - but non blocking on my end.

File: cmd/account.go (hunk 1 above)
Comment (btoonk) [reply cmd/account.go:131]: We have the API package that we should use for this.

File: internal/connector/core/core.go [hunk 3]
```diff
@@ -332,6 +314,8 @@ func (c *ConnectorCore) CheckSocketsToDelete(ctx context.Context, socketsFromApi
 					return err
 				}
 				localSocketsMap[apiSocket.ConnectorData.Key()] = *createdSocket
+				delete(socketApiMap, apiSocket.ConnectorData.Key())
```
[the remark this answers is not in the document]
Comment (btoonk) [reply internal/connector/core/core.go:317]: Not sure, I think the next line is wrong. We should use `createdSocket.ConectorData.Key()`

**Review by th3wingman — APPROVED** [review]
Tested and deployed into staging

**Pull request merged**